workflows/mcmicro.nf

```diff
@@ -63,78 +65,170 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
+                .set { ashlar_input_keyed }
+
+                ch_from_samplesheet.ashlar_input
+                    .transpose()
+                    .set { ch_basicpy_input }
+
+            BASICPY(ch_basicpy_input)
+            ch_versions = ch_versions.mix(BASICPY.out.versions)
+
+            BASICPY.out.fields
+                .transpose()
+                .map { [[it[1].getBaseName()[0..-5],it[0]], it[1]]}
+                .groupTuple()
+                .set { correction_files_keyed }
+
+            ashlar_input_keyed
+                .concat(correction_files_keyed)
+                .groupTuple()
+                .map { [it[0][1], it[1][1]] }
+                .transpose()
+                .branch {
+                    dfp: it =~ /-dfp.tiff/
+                    ffp: it =~ /-ffp.tiff/
+                }
+                .set { ordered_correction_files }
+            ch_dfp = ordered_correction_files.dfp
+                .groupTuple()
+                .map { it[1] }
+            ch_ffp = ordered_correction_files.ffp
+                .groupTuple()
+                .map { it[1] }
+
+        } else if(params.illumination == 'manual') {
+
+            dfp_index = sample_sheet_index_map["dfp"]
+            ffp_index = sample_sheet_index_map["ffp"]
+
+            if (input_type == "cycle") {
+                samplesheet = "input_cycle"
+            } else if (input_type == "sample") {
+                samplesheet = "input_sample"
+            }
+            ch_manual_illumination_correction = Channel.fromSamplesheet(
+                samplesheet,
+                parameters_schema: parameters_schema,
+                skip_duplicate_check: false
+            )
+            .multiMap
+                { it ->
+                    dfp: it[dfp_index]
+                    ffp: it[ffp_index]
+                }
+
+            ch_dfp = ch_manual_illumination_correction.dfp
+            ch_ffp = ch_manual_illumination_correction.ffp
+        }
+
+    } else {
+        ch_dfp = []
+        ch_ffp = []
+    }
 
-    //     ch_dfp = ch_tif.filter { file -> file.name.endsWith('.dfp.tiff') }
-    //     ch_ffp = ch_tif.filter { file -> file.name.endsWith('.ffp.tiff') }
-    // }
-    // */
+    INPUT_CHECK( input_type, params.input_sample, params.input_cycle, params.marker_sheet )
 
-    ASHLAR(raw_images, dfp, ffp)
+    // ASHLAR(ch_from_samplesheet.ashlar_input, [], [])
+    ASHLAR(ch_from_samplesheet.ashlar_input, ch_dfp, ch_ffp)
     ch_versions = ch_versions.mix(ASHLAR.out.versions)
 
     // // Run Background Correction
     // BACKSUB(ASHLAR.out.tif, ch_markers)
-    // ch_versions = ch_versions.mix(BACKSUB.out.versions)
+    //BACKSUB(ASHLAR.out.tif, [[id: "backsub"], params.marker_sheet])
+    //ch_versions = ch_versions.mix(BACKSUB.out.versions)
+
+    // Run Segmentation
 
-    // // Run Segmentation
     DEEPCELL_MESMER(ASHLAR.out.tif, [[:],[]])
     ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
 
```

Comment:
perhaps add comment, what if we want a different segmentation method, CELLPOSE comes to mind

Reply:
The final goal is to run multiple/all segmentors in parallel and quantify in parallel using the masks from each one.